### ISSUE TITLE
Stocks-Shares revision/concept

### ISSUE THREAD

**Issue by renato-mmex** [moneymanagerex/moneymanagerex#4793]
**Stocks-Shares revision/concept**

For some time now, it has been written again and again that the Stocks-Shares program section needs to be revised.

I would like to start a discussion to get a concept going. As many users as possible, of course especially those who use this program part and have knowledge in this matter, should contribute with their ideas.

The big challenge will be, in my opinion, the correct handling of foreign currencies.

I start with the functions that I think are needed:

Stocks processing
 -------------------
1.   Purchase               
2.   Sale            
3.   Entitlements             
4.   Coupon/dividends    
5.   Payment             
6.   Repayment           
7.   Split
8.   Exchange               
9.   Stock dividend        
10. Closing              
11. Takeover              

                  

By the way: about 40 years ago I added (programmed) such a solution to my dialog financial accounting (written in PLB 1975), only my detailed knowledge has dwindled a bit in the meantime.  If my English sounds a bit funny, it is thanks to www.deepl.com.

**Comment by vomikan** [comment]
https://github.com/moneymanagerex/moneymanagerex/issues/2702
https://github.com/moneymanagerex/moneymanagerex/issues/2656
https://github.com/moneymanagerex/moneymanagerex/issues/2147

**Comment by whalley** [comment]
I've always been a bit perplexed by the complexity of the investment and shares accounts and the seeming 1:1 relationship. I liked the simplicity of the way AceMoney did this.

Similar to MMEX, AceMoney holds an Investment account.
![CleanShot 2022-10-18 at 23 56 52](https://user-images.githubusercontent.com/487345/196560462-2b787c7a-a35b-4aff-8629-b88ea81a65b9.png)

However you are not forced to create a linked Share account you can add an investment transaction from any account. When creating the investment it uses actions like Buy/Sell rather that the withdraw/deposit that MMEX confusing uses and does not clutter the interface with Payee and Category. My personal preference would be when creating investment transactions to limit the account list to the shares accounts to avoid confusion.

![CleanShot 2022-10-18 at 23 56 45](https://user-images.githubusercontent.com/487345/196560878-d7eb488f-9ffc-4e40-a121-67f9661be122.png)

The way I use investment platform suits this well as I have a number of investments with a broker platform and a single CASH account used to transfer money to/form checking accounts and to make investment purchases on.

**Comment by whalley** [comment]
What exactly is the purpose of the Share account that is (optionally) created when a new stock investment is created? Why would we want a share account for every stock? I tend to have share accounts for cash accounts I have at a broker/shares platform that is use to purchase/sell stocks.

Would it be better to have a 'default' share account associated with the investment account?

**Comment by whalley** [comment]
> Possible that you have different investment strategies. Hence may be useful in such cases to have the strategies in different accounts, with all their trasncations.

Agreed, but at present it assumes you will create (prompts the user) a single share account per-stock investment. I'm suggesting that we remove that and allow the user to associate transaction to a share account as needed. Needs a bit more analysis.

**Comment by zeezhao** [comment]
I agree that this should be up to the user to decide which account, and should not prompt.

**Comment by whalley** [comment]
So interesting words/suggestions in prior posts...
https://github.com/moneymanagerex/moneymanagerex/issues/336#issuecomment-135741409
https://github.com/moneymanagerex/moneymanagerex/issues/336#issuecomment-135744981

Think it would be good to map out any revised proposal in a document similar to this...
https://docs.google.com/document/d/1t19q4K5TOvh-pHQClDXno4zOTNNBPJi8jhuthSmjpd4/edit#heading=h.lps9s630b20h

**Comment by nickallington** [comment]
I am doing some work on specifying changes in this area. Part of the confusion (as I see it) is between what is an 'account' and what isn't, and what the purpose of the accounts is. The Share account should be created automatically from a purchase transaction, and the purchase transaction should result in a Holding, which is displayed as part of a Portfolio. The Share account is only for the purpose of recording income and costs against the shares - so, dividends, and any taxes and fees associated with the shares themselves. The net balance of any dividend is then transferred from the Share account to a user account, either directly, or via a broker account. One of the things that is missing is the ability to show a broker funds holding account within a portfolio, which it is essentially a part of since it exists to facilitate share purchases via that broker.

It would be possible in theory to hold all the purchase, dividend and sale transactions in the same account, but I am not sure that makes for easy account viewing. For me, the Porfolio approach, showing Holdings, and a Share account showing share transactions, makes sense, but it could perhaps be implemented more clearly.

**Comment by nickallington** [comment]
Suggest to keep open while Investment rewrite is pending.
